packages/expo/src/useOAuth.ts

```diff
@@ -0,0 +1,84 @@
+import { useSignIn, useSignUp } from '@clerk/clerk-react';
+import type { OAuthStrategy } from '@clerk/types';
+import * as AuthSession from 'expo-auth-session';
+import * as WebBrowser from 'expo-web-browser';
+
+export type UseOAuthFlowParams =
+  | {
+      strategy: OAuthStrategy;
+      redirectUrl?: string;
+    }
+  | OAuthStrategy;
+
+export function useOAuth(params: UseOAuthFlowParams) {
+  let strategy: OAuthStrategy;
+  let redirectUrl: string;
+
+  if (typeof params === 'string') {
+    strategy = params;
+  } else if (typeof params === 'object') {
```

Comment:
Just to be extra safe we can also check for:
```suggestion
  } else if ('strategy' in params)  {
```

Also, what happens if `params` is missing? Are we throwing further down the flow?

Reply:
We probably need to create a helper to handle validation + type inference for overloaded functions and params. Something like a mini-zod implementation tailored to our use cases.

Reply:
Fixed. I've also just kept the object although I do like the flexibiity of passing just a strategy.